src/compiler/scala/tools/nsc/settings/StandardScalaSettings.scala

```diff
@@ -71,15 +71,17 @@ trait StandardScalaSettings { _: MutableSettings =>
       if (releaseValue.map(_.toInt < setting.value.toInt).getOrElse(false))
         errorFn("-release cannot be less than -target")
       if (!setting.deprecationMessage.isDefined)
-        if (SupportedTargetVersions.contains(setting.value))
-          setting.withDeprecationMessage("Use -release instead to compile against the correct platform API.")
-        else {
+        if (setting.value.toInt > MaxSupportedTargetVersion) {
+          setting.withDeprecationMessage(s"Scala 2.12 cannot emit valid class files for targets newer than $MaxSupportedTargetVersion (this is possible with Scala 2.13). Use -release to compile against a specific platform API version.")
+          setting.value = DefaultTargetVersion
+        } else if (setting.value.toInt < MinSupportedTargetVersion) {
           setting.withDeprecationMessage(s"${setting.name}:${setting.value} is deprecated, forcing use of $DefaultTargetVersion")
-          setting.value = DefaultTargetVersion   // triggers this hook
+          setting.value = DefaultTargetVersion
         }
     }
     .withAbbreviation("--target")
-  def targetValue: String = releaseValue.getOrElse(target.value)
+  // Unlike 2.13, don't use `releaseValue.getOrElse(target.value)`, because 2.12 doesn't have a fix for scala-dev#408
+  def targetValue: String = target.value
```

Comment:
If they use just `-release` then they get default target 8. If they `-release 8 -target 6` they will get the existing deprecation and target 8. If they are building on JDK 8, they can only use `-release 8` or use `-target`. The changed behavior is `-release 11` where they get target 8 as a limitation with no warning to fail a build. Only explicit `-target 9` would warn.